src/main/java/de/tum/cit/aet/artemis/iris/service/session/IrisLectureChatSessionService.java

```diff
@@ -0,0 +1,150 @@
+        this.authCheckService = authCheckService;
+    }
+
+    @Override
+    public void sendOverWebsocket(IrisLectureChatSession session, IrisMessage message) {
+        irisChatWebsocketService.sendMessage(session, message, null);
+    }
+
+    // This is the message from the user sent to Iris
+    @Override
+    public void requestAndHandleResponse(IrisLectureChatSession lectureChatSession) {
+        var session = (IrisLectureChatSession) irisSessionRepository.findByIdWithMessagesAndContents(lectureChatSession.getId());
+        var lecture = lectureRepository.findByIdElseThrow(session.getLecture().getId());
+        var course = lecture.getCourse();
+
+        if (!irisSettingsService.isEnabledFor(IrisSubSettingsType.LECTURE_CHAT, course)) {
+            throw new ConflictException("Iris is not enabled for this lecture", "Iris", "irisDisabled");
+        }
+
+        var conversation = session.getMessages().stream().map(PyrisMessageDTO::of).toList();
+        pyrisPipelineService.executePipeline("lecture-chat", "default",
+                pyrisJobService.createTokenForJob(token -> new LectureChatJob(token, course.getId(), lecture.getId(), session.getId())),
+                dto -> new PyrisLectureChatPipelineExecutionDTO(new PyrisCourseDTO(course), conversation, new PyrisUserDTO(session.getUser()), dto.settings(), dto.initialStages()),
+                stages -> irisChatWebsocketService.sendMessage(session, null, stages));
+    }
+
+    @Override
+    public void checkHasAccessTo(User user, IrisLectureChatSession session) {
+        if (!session.getUser().equals(user)) {
+            throw new AccessForbiddenException("Iris Lecture chat Session", session.getId());
+        }
+
+        authCheckService.checkHasAtLeastRoleForLectureElseThrow(Role.STUDENT, session.getLecture(), user);
+    }
+
+    /**
+     * Checks if the user has access to the Iris session.
+     * A user has access if they have access to the lecture and the session belongs to them.
+     * If the user is null, the user is fetched from the database.
+     *
+     * @param user    The user to check
+     * @param session The session to check
+     * @return weather the user has access to the session
+     */
+    public boolean hasAccess(User user, IrisLectureChatSession session) {
+        try {
+            checkHasAccessTo(user, session);
+            return true;
+        }
+        catch (AccessForbiddenException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Handles the status update of a lecture chat job.
+     *
+     * @param job          The job that is updated
+     * @param statusUpdate The status update
+     * @return The same job that was passed in
+     */
+    public LectureChatJob handleStatusUpdate(LectureChatJob job, PyrisLectureChatStatusUpdateDTO statusUpdate) {
+        // TODO: LLM Token Tracking - or better, make this class a subclass of AbstractIrisChatSessionService
```

Comment:
As CodeRabbit suggested, it would be great if you could add the token tracking to this PR, as you already sent the tokens from Pyris to Artemis and removed the TODO